File: Other/C/Cracker.cpp
// Copyright (c) Alexandre-Xavier Labonté-Lamoureux, 2017

#include <iostream>
#include <string>
using namespace std;

string chars = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";

void Generate(const size_t& l, string s)
{
	if (l == 0) {
		cout << s << endl;
		return;
	}

	for (size_t i = 0; i < chars.size(); i++)
		Generate(l - 1, s + chars[i]);
}

void Crack(const size_t& pl)
{
	size_t length = 1;
	while (length <= pl) {
		Generate(length, "");
		length++;
	}
}

int main(int argc, char* argv[])
{
	if (argc >= 3)
		chars = argv[2];

	if (argc >= 2)
		Crack(atoi(argv[1]));
	else
		cout << "cracker [length] [chars]" << endl;
}

File: Other/Python/Cracker.py
#!/usr/bin/env python
# Copyright (c) Alexandre-Xavier Labonte-Lamoureux, 2017

import sys

chars = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";

def Generate(length, word):
	if (length == 0):
		print(word)
		return

	for i in range(len(chars)):
		Generate(length - 1, word + chars[i])

def Crack(pl):
	length = 1
	while (length <= pl):
		Generate(length, "")
		length += 1

def Main():
	if (len(sys.argv) >= 3):
		global chars
		chars = sys.argv[2]

	if (len(sys.argv) >= 2):
		Crack(int(sys.argv[1]))
	else:
		print("cracker [length] [chars]")

if __name__ == "__main__":
	Main()
